bokehjs/src/lib/models/widgets/tables/cell_formatters.ts

```diff
@@ -273,7 +273,8 @@ export class DateFormatter extends StringFormatter {
     const {nan_format} = this
     value = isString(value) ? parseInt(value, 10) : value
     let date: string
-    if ((value == null || isNaN(value)) && nan_format != null)
+    // Handle null, NaN and NaT
+    if ((value == null || isNaN(value) || value === -9223372036854776) && nan_format != null)
```

Comment:
Where does this value come from, because I get:
```
In [1]: import pandas as pd                                                                                                                  

In [2]: pd.NaT.value                                                                                                                         
Out[2]: -9223372036854775808
```
which btw. is outside of safe integer range in JS.

Reply:
>is outside of safe integer range in JS.

Presumably that's why it's `-9223372036854776` in JS. I logged the value that the formatter sees when serializing NaT.

Reply:
Well:
```
$ node
> -9223372036854775808
-9223372036854776000
```

Reply:
Can't tell you where the discrepancy comes from but if you serialize a DataFrame with NaTs in it that's the value that arrives in JS.

Reply:
Maybe something is off in the deserialization? The difference between `-9223372036854776000` and `-9223372036854776` are just the dropped zeros at the end.

Reply:
Unless we have to differentiate between `NaN` and `NaT` on JS side, the simplest solution would be to just serialize `NaT`s as `NaN`s.

Reply:
>Unless we have to differentiate between NaN and NaT on JS side, the simplest solution would be to just serialize NaTs as NaNs.

How would that work for integer arrays? In fact I'm somewhat confused how the date array gets serialized to buffer and is deserialized into a typed array, isn't it int64? I know date arrays **used to** get sent as raw JSON rather than bytes but that seems to have changed.

Reply:
Serializer does reduce the resolution (`/1000`) to avoid dealing with 64-bit numbers. Nevertheless, if we proceed with supporting `NaT` via a constant, then this needs to be acknowledged and tested on both sides of the serialization scheme.

Reply:
> isn't it int64?

It's encoded as 53-bit sudo integers (float64 arrays) (with reduced resolution). 64-bit integers and integer arrays are not yet widely supported enough for us to use.

Reply:
>Serializer does reduce the resolution (/1000) to avoid dealing with 64-bit numbers.

That explains the difference between `-9223372036854776000` and `-9223372036854776` then. Agree that we should explicitly test NaT handling though.

>It's encoded as 53-bit sudo integers (float64 arrays) (with reduced resolution). 64-bit integers and integer arrays are not yet widely supported enough for us to use.

Right, my confusion was about the fact that it's int64 on the Python end and how that gets translated to JS since I distinctly remember this not being supported and always falling back to regular JSON serialization in the past.

Fully agree that we'll need to handle NaTs more generally in the serialization, looks like plots aren't happy about NaTs right now:

![Screen Shot 2020-08-31 at 11 14 54 AM](https://user-images.githubusercontent.com/1550771/91703986-4c30e900-eb7b-11ea-8c9c-8bf1a3c7d397.png)

Reply:
This isn't a new issue, more discussion is available in #10379.